### PR TITLE
user auth flow

### PR DESCRIPTION
**Pull request by SophieDeBenedetto**
This is can probably be refactored a bit but wanted to get the PR up to make sure we're comfortable with this direction: 

* No persistence of users
* "Authentication" consists of checking for username in params and setting it in session

This PR introduces the following behavior:

As an un-authenticated user
When I visit the root path
I see a welcome page and link to log in
When I click the link to log in, I am taken to `/login` where I can "log in" with _just_ a username 
When I fill out the username field and click "log in"
Then I am redirect to the `/cards` page and I see the header includes a greeting, `"Hi, #{my_username}"` and a link to log out

When I do NOT fill out the username field and click "log in"
Then the login page re-renders and I see an error message that the username field cannot be blank

As authenticated user
When I visit the root path
Then I see a welcome page with a link the header to log out
When I click "log out"
I am redirected to the `/login` path 

As an authenticated user
When I visit `/cards`
Then I see the first card's title and the header includes a link to log out

### CODE REVIEW

File: lib/pointing_party_web/controllers/session_controller.ex [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  use PointingPartyWeb, :controller
+  alias PointingParty.User
+
+  def new(conn, _params) do
+    changeset = User.changeset(%User{})
+    render(conn, "new.html", changeset: changeset)
+  end
+
+  def create(conn, params) do
+    changeset = User.changeset(%User{}, params["user"])
```
Comment (SophieDeBenedetto) [inline lib/pointing_party_web/controllers/session_controller.ex]: Might be nice to refactor this a bit to hide the calls to `User.changeset`, `changeset.valid?` and `Ecto.Changeset.apply_changes`, maybe wrap them up in some helper functions in the User module?

File: lib/pointing_party_web/controllers/session_controller.ex [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    if changeset.valid? do
+      user =  Ecto.Changeset.apply_changes(changeset)
+      put_session(conn, :username, user.username)
```
Comment (SophieDeBenedetto) [inline lib/pointing_party_web/controllers/session_controller.ex]: Could refactor this into a function like, `login` or something more descriptive

**Comment by SophieDeBenedetto** [comment]
I added an *extremely* DIY logo for now, just to have something fun to look at that isn't the "Phoenix Framework" image.

File: lib/pointing_party_web/controllers/session_controller.ex [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  alias PointingParty.Account
+
+  def new(conn, _params) do
+    changeset = Account.changeset(%Account{})
+    render(conn, "new.html", changeset: changeset)
+  end
+
+  def create(conn, params) do
+    case Auth.login(params["account"]) do
+      {:ok, account} ->
```
Comment (doomspork) [inline lib/pointing_party_web/controllers/session_controller.ex]: What do you think of using pattern matching to pull out the username and cleaning up the pipeline a tad?

```elixir
def create(conn, params) do
  case Auth.login(params["account"]) do
    {:ok, %{username: username}} ->
      conn
      |> put_session(:username, username)
      |> redirect(to: "/cards") 
      |> halt()

    {:error, changeset} ->
      render(conn, "new.html", changeset: changeset)
  end
end
```

File: lib/pointing_party_web/plugs/auth.ex [hunk 4]
```diff
@@ -0,0 +1,17 @@
+defmodule PointingPartyWeb.Plugs.Auth do
+  import Plug.Conn
+  import Phoenix.Controller
+
+  def init(default), do: default
+
+  def call(conn, _default) do
+    case authenticate(conn) do
+      nil -> redirect(conn, to: "/login") |> halt()
```
Comment (doomspork) [inline lib/pointing_party_web/plugs/auth.ex]:
```elixir
conn |> redirect(to: "/login") |> halt()
```

File: lib/pointing_party_web/views/layout_helpers.ex [hunk 5]
```diff
@@ -0,0 +1,10 @@
+defmodule PointingPartyWeb.LayoutHelpers do
+  use Phoenix.HTML
+
+  @doc """
+  Generates tag for inlined form input errors.
+  """
+  def signed_in?(conn) do
```
Comment (doomspork) [inline lib/pointing_party_web/views/layout_helpers.ex]: The `?` here makes me thing this is expected to return a boolean, but the `conn.assigns[:username]` will return `nil` or a value.

What do you think about changing this to:

```elixir
def signed_in?(conn) do
  not is_nil(conn.assigns[:username])
end
```

File: lib/pointing_party_web/controllers/session_controller.ex [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+  def new(conn, _params) do
+    changeset = Account.changeset(%Account{})
+    render(conn, "new.html", changeset: changeset)
+  end
+
+  def create(conn, params) do
+    case Auth.login(params["account"]) do
+      {:ok, %{username: username}} ->
+        put_session(conn, :username, username)
```
Comment (doomspork) [inline lib/pointing_party_web/controllers/session_controller.ex]: One more small change here, can we start this pipeline with a raw value (`conn`)?

```elixir
conn
|> put_session(...)
|> redirect(...)
|> halt()
```

**Pull request merged**